test/promise-component-mocha.js

```diff
@@ -0,0 +1,149 @@
+// promise-component-mocha.js
+
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.should();
+chai.use(chaiAsPromised);
+
+var _ = require('underscore');
+var noflo = require('noflo');
+var promiseComponent = require('../components/promise-component.js');
+
+describe('promise-component', function() {
+    it("should reject undefined definition", function() {
+        return Promise.resolve().then(promiseComponent).should.be.rejected;
+    });
+    it("should reject empty definition", function() {
+        return Promise.resolve({}).then(promiseComponent).should.be.rejected;
+    });
+    it("should reject empty inPorts", function() {
+        return Promise.resolve({inPorts:{}}).then(promiseComponent).should.be.rejected;
+    });
+    it("should reject empty resolvePort", function() {
+        return Promise.resolve({inPorts:{in:{}},resolvePort:{}}).then(promiseComponent).should.be.rejected;
+    });
+    it("should reject empty rejectPort", function() {
+        return Promise.resolve({inPorts:{in:{}},rejectPort:{}}).then(promiseComponent).should.be.rejected;
+    });
+    it("should trigger ondata function", function() {
+        var handler;
+        return Promise.resolve({
+            inPorts:{
+                'in':{
+                    ondata: function(payload) {
+                        handler(payload);
+                    }
+                }
+            }
+        }).then(promiseComponent).then(createComponent).then(function(component){
+            return new Promise(function(callback){
+                handler = callback;
+                sendData(component, 'in', "hello");
+            });
+        }).should.become("hello");
```

Comment:
Let's talk about this.  I think I am still missing something (I followed what the comment says before), but it's probably more a question for me than the code here.